app/views/business_categories/show.html.haml

```diff
@@ -21,11 +21,11 @@
             %th= t('companies.index.company_number')
             %th
       %tbody
-        - @business_category.companies.distinct.sort {|cp1, cp2| cp1.name <=> cp2.name}.each do |company|
-          - if company_complete? company
-            %tr.company
-              %td= link_to company.name, company
-              %td= company.region.name
-              - if current_user.try(:admin)
-                %td= company.company_number
-                %td= link_to t('edit'), edit_company_path(company)
+        - complete_cos = @business_category.companies.distinct.select{|c| c.complete? }
+        - complete_cos.sort {|cp1, cp2| cp1.name <=> cp2.name}.each do |company|
```

Comment:
Suggest changing business_categories_controller#show from this:
```
def show
  @companies = Company.all
end
```
to this:
```
def show
  @companies = @business_category.companies.complete.order(:name)
end
```
Then these two lines in the view become:
```
- @companies.each do |company|
```
This moves the logic to the controller, and leverages the scope added in the company model.